working/augmentation-libraries/feature-specification.md

```diff
@@ -356,6 +356,39 @@ augmented, but it follows generally the same rules as any normal identifier:
 In all relevant cases, if the augmented member is an instance member, it is
 invoked with the same value for `this`.
 
+Assume that the identifier `augmented` occurs such that the outermost
+enclosing declaration is not an augmenting declaration. In this case, the
+identifier is taken to be a reference to a declaration which is in scope.
+
+*In other words, `augmented` is just a normal identifier when it occurs
+anywhere other than inside an augmented declaration.*
+
+*Note that, for example, `augmented()` is an invocation of the augmented
+function or method when it occurs in an augmenting function or method
+declaration. (In the latter case, the augmenting method declaration must
+occur inside an augmenting type-introducing declaration, e.g., an
+augmenting class or mixin declaration). This is also true if `augmented()`
+occurs inside a local function declaration inside the body of that function
+or method declaration. We could say that `augmented` is a contextual
+keyword because it is unable to refer to a declaration in scope when it
+occurs inside an augmenting declaration, it always has the special meaning
+which is associated with augmentations.*
+
+A compile-time error occurs if a declaration with the name `augmented`
+occurs in a location where the outermost enclosing declaration is
+augmenting. *This error is applicable to all such declarations, e.g.,
+local functions, local variables, parameters, and type parameters.*
+
+A compile-time error occurs if `augmented` occurs in a non-augmenting
```

Comment:
This one doesn't seem absolutely necessary to me - it only really needs to be an error if it is inside an augmenting declaration which introduces a meaning for `augmented` - in other words using `augmented()` inside an `augment class C` declaration is safe to do.

But, I can see that it might be hard to specify (and confusing to understand). So I am ultimately OK with specifying it this way as a simplification, cc @munificent thoughts?

Reply:
```dart
abstract interface class I {
  int bar();
}

class C implements I {}

augment class C {
  int bar() {
    return augmented(); // Not safe
  }
}
```

Reply:
(there is no augmented function, since this isn't an augmenting method, so it is just an identifier called `augmented`)

Reply:
> Assuming that there is a declaration augmented in the surrounding scope, why can that not just refer to it?
I didn't think about it. Right

Reply:
We talked briefly in person and the intention here is just to simplify the rules so they are easier to understand, which I think is well motivated 👍

Reply:
I suggested that we should make `augmented` refer to the augmentation mechanism rather than any declared name in scope _everywhere_ in an `augment` declaration because I thought that would be the most readable approach.

I do understand that it would be possible for a non-augment declaration inside an augment declaration to, say, call a top level function named `augmented`, but I thought it would be really tricky for a human reader of the code to understand that this is what we're doing. So, pure readability.